tsl/src/compression/api.c

```diff
@@ -1017,6 +1017,7 @@ tsl_get_compressed_chunk_index_for_recompression(PG_FUNCTION_ARGS)
 {
 	Oid uncompressed_chunk_id = PG_ARGISNULL(0) ? InvalidOid : PG_GETARG_OID(0);
 	Chunk *uncompressed_chunk = ts_chunk_get_by_relid(uncompressed_chunk_id, true);
+	Chunk *compressed_chunk = ts_chunk_get_by_id(uncompressed_chunk->fd.compressed_chunk_id, true);
```

Comment:
This lookup is expensive, is it enough to get the relid here? There's a less heavy function `ts_chunk_get_relid` for that, or maybe `ts_chunk_get_formdata`.

Same comment for another usage below.

Reply:
Fixed as you suggested. Help me understand why ```ts_chunk_get_by_id``` is a costly operation ?

```ts_chunk_get_by_id``` does following:
1. Build scankey on chunk id.
2. Take AccessShareLock on CHUNK catalog table.
3. Scan the index and fetch the required row.

What you suggested is:
1. Call ```ts_chunk_get_relid``` to get chunk relid.
2. Call ```ts_chunk_get_by_relid``` to get the chunk from relid.

```ts_chunk_get_by_relid``` does following:
1. Build scankey on schem, chunk name.
2. Take AccessShareLock on CHUNK catalog table.
3. Scan the index and fetch the required row.